### ISSUE TITLE
ICA Youth Network data - IDs, weirdness

### ISSUE THREAD

**Issue by wu-lee** [DigitalCommons/open-data#22]
@ColmMassey - I seem to remember you said there was some new data for the ICA Youth Network? I'm hoping to get data with the coop's ID incorporated, so we don't have to make that up (and maintain the ID mapping if new data arrives).

Also, there are some oddities in the data.

Firstly, what I know about. I had a look on SEA Nextcloud, and found these:

## `Youth-ledCoops.csv`

First two lines are:

```
Organization Type,Name,name,Region,Country,City,Latitude,Longitude,Size,Type,Sector,Address,Description,Additional Details,Website,Email
Youth-led Co-ops,Agricultores Federados Argentinos S.C.L.,Cooperativa de producción,Americas,Argentina,Rosario,-32.9442426,-60.6505388,100+,Cooperativa de producción,Agriculture,Dto Educación y Capacitació Mitre 1132 Rosario S2000COX Argentina,"AFA SCL es el mas fuerte originador de granos del pais, con una densidad logistica unica de mas de 140 plantas de servicio agropecuario.","Agricultores Federados Argentinos S.C.L. es la principal Cooperativa Agropecuaria de primer grado en Argentina y una de las más grandes y pujantes de América Latina. Los 36.000 Productores Asociados cuentan con una estructura compuesta por más de 1.600 empleados permanentes en 26 Centros Cooperativos Primarios, que sumados a los Sub-Centros, Oficinas y Representaciones permiten que la entidad tenga presencia arraigada en 130 localidades de 9 provincias Argentinas: Santa Fe, Buenos Aires, Córdoba, Entre Ríos, Santiago del Estero, Chaco, Salta, Tucumán y San Luis. La inmensa extensión territorial genera un acopio del orden de las 5.000.000 de Toneladas anuales, una capacidad de almacenaje de 3.000.000 de Toneladas y una fuerte logística en transporte compuesta por más de 220 camiones propios. Esta extensa red permite brindar una diversa gama de servicios durante todo el año y otorgar a los Asociados importantes beneficios económicos y sociales. En los últimos años, en pos del crecimiento, la innovación y la búsqueda de mayores beneficios para los Asociados, AFA S.C.L. priorizó proyectos orientados al agregado de valor de la producción primaria, profundizando procesos de industrialización de materias primas y de agroinsumos, como así también generando nuevos servicios de acopio y comercialización de productos especiales que ayuden a la diversificación productiva de los Asociados.",http://www.afascl.com,jovenes@afascl.coop
Youth-led Co-ops,Coop. CASMUL,Cooperativa de producción,Americas,Honduras,"San Manuel, Cortes",15.3327467,-87.9187711,100+,Cooperativa de producción,Agriculture,"504 San Manuel, Cortes, Honduras","Mi coop. Es de producción, exportamos banano de primera a los Estados Unidos, sembrados palma africana la cual vendemos la fruta a una coop. Encargada de extraer el aceite y derivados.",,http://casmul.blogspot.com,n/a
```
So, problems here are:
- No ID.
- Duplicate `name` field. (One is called `Name`, the other `name`. The latter seems to be something else, containing "Cooperativa de producción" which is repeated in field 10, `Type`, so this is presumably a duplicate)

And in our conversion, we concatenate `Description` and `Additional Details` into `Description`. This is fine, but reading the result, it to me it looks like there should be a paragraph break between the two. For now I tried concatenating these with a `<p>` tag which works in the HTML version, but I'm not so sure works in general, i.e. in RDF and TTL.

So this raises two questions, which might deserve their own issues but I'm mention here:

- What format are fields like this in general?

I think we should be clear in our specifications about whether it's text, HTML, Markdown or something else, and also what characters are allowed. Specifically, are linefeeds valid? Or will they just get lost? (As they do when embedding verbatim in HTML, which ignores linefeeds). What encoding is used? What is a reasonable length, i.e. how verbose should they be? etc. Then we can make sure this gets rendered correctly / legibly.

Then there's this:

- Our current conversion to HTML is vulnerable to code-injection exploits.

Not that we expect incoming data to contain malicious, or even accidental `<script>` tags, tracking links, `<iframes>` or other strangeness, but with data coming from web forms or other un-moderated sources we can't vouch for we'd be wise to make it hard to play games like that. 

Even ignoring that possibility, it's probably a good idea to (for example) escape HTML markup if it's not meant to be there, or it won't always be valid / render correctly. As it is, we insert text into HTML verbatim and unescaped, so angle brackets &lt;like this&gt; will disappear, and line-feeds and other formatting will get lost, the encoding may not match and resulting in ["Mojibake"][1] garbage, etc. etc.

[1]: https://en.wikipedia.org/wiki/Mojibake

If we know what format text is in, we can ensure it survives the trip to the final rendered form in a way which still makes sense.

## `20200406-Youth-led Co-ops.csv`

This seems to be more recent. But doesn't seem to have an ID either. Plus, it appears to be invalid, or at least a bit scrambled. The first couple of rows of the file:

```
"Organization Type,Name,name,Region,Country,City,Latitude,Longitude,Size,Type,Sector,Address,Description,Additional Details,Website,Email,.coop,,,",,
"Youth-led Co-ops,Coop. CASMUL,Cooperativa de produccin,Americas,Honduras,""San Manuel, Cortes"",15.3327467,-87.9187711,100+,Cooperativa de produccin,Agriculture,""504 San Manuel, Cortes, Honduras"",""Mi coop. Es de produccin, exportamos banano de primera a los Estados Unidos, sembrados palma africana la cual vendemos la fruta a una coop. Encargada de extraer el aceite y derivados."",,http://casmul.blogspot.com,n/a,FALSE,,,",,
```

So, there are quotes around almost the entire first 20 or so fields, resulting in the CSV parser seeing just three. This doesn't just seem to be the first line either.

Overall, this data looks like it's been exported badly, and we should ask for something sensible.

**Comment by ColmMassey** [comment]
Why are you working on this?

**Comment by ColmMassey** [comment]
There is no point working on the ica-youth data until the new data comes through.

**Comment by wu-lee** [comment]
So, I'm not actively working on this right now. The ica-youth-network project just happened to be what @dtmakm27  de-makefile'd first, and is therefore a kind of pilot case which I've looked at in detail.

I've created this ticket because I don't want to forget them, I can't see anywhere else to put them, and I think the points will affect all the other project data. I agree we need more data, but I wasn't sure and I where we stood there.
